api/renter.go

```diff
@@ -242,8 +242,25 @@ func (srv *Server) renterUploadHandler(w http.ResponseWriter, req *http.Request,
 // renterHostsActiveHandler handes the API call asking for the list of active
 // hosts.
 func (srv *Server) renterHostsActiveHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	hosts := srv.renter.ActiveHosts()
+	if req.FormValue("numhosts") == "" {
+		writeJSON(w, ActiveHosts{
+			Hosts: hosts,
+		})
+		return
+	}
+
+	var numHosts uint64
+	_, err := fmt.Sscan(req.FormValue("numhosts"), &numHosts)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if numHosts > uint64(len(hosts)) {
+		numHosts = uint64(len(hosts))
+	}
```

Comment:
I'd prefer to see these two cases combined, such that there's only one `writeJSON`. `numhosts == ""` should be a special case that sets `numHosts` to `len(hosts)`.